spec/reek/smells/attribute_spec.rb

```diff
@@ -17,6 +17,47 @@
       ctx = Reek::Core::CodeContext.new(nil, Reek::Source::SourceCode.from(src).syntax_tree)
       expect(@detector.examine_context(ctx)).to be_empty
     end
+
+    shared_examples_for 'no attribute found' do
+      before :each do
+        ctx = Reek::Core::CodeContext.new(nil, Reek::Source::SourceCode.from(@src).syntax_tree)
+        @smells = @detector.examine_context(ctx)
+      end
+
+      it 'records no attribute' do
+        expect(@smells.length).to eq(0)
```

Comment:
yes, the attribute spec is pretty weird. i thought about refactoring the spec. i digged into the rabbit hole and stopped half way through. i'm much in favour of redoing the whole spec with the new technique you described.

i'll look into preparing another PR for the spec.